client.go

```diff
@@ -147,26 +153,48 @@ func (c *httpClient) doRequest(ctx context.Context, method, url string, params R
 	// Apply mutators
 	for _, mutate := range mutators {
 		if err := mutate(req); err != nil {
+			if c.debug && c.logger != nil {
+				c.logger.Errorf("failed to mutate request: %v", err)
+			}
 			return fmt.Errorf("failed to mutate request: %w", err)
 		}
 	}
 
+	// Log the request if in debug mode
+	if c.debug && c.logger != nil {
+		c.logger.Debugf("sending request: %s %s", method, url)
```

Comment:
Could we also show the request body and headers here and in the response log?

Reply:
Thanks for the feedback! I'll definitely look at text/template.

Reply:
Fixed!